File: BZSwift/0223Test/0223Test/ViewController.m
//
//  ViewController.m
//  0223Test
//
//  Created by ZBin on 16/2/22.
//  Copyright © 2016年 BZ. All rights reserved.
//

#import "ViewController.h"
#import "UITextView+Extension.h"
@interface ViewController (){
    NSAttributedString *attributedString;
}
@property (weak, nonatomic) IBOutlet UILabel *labelText;
@property (weak, nonatomic) IBOutlet UITextView *textView;
@property (weak, nonatomic) IBOutlet UITextField *textField;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSString *String = [[NSString alloc] initWithString:self.labelText.text];
    attributedString = [[NSMutableAttributedString alloc] initWithString:String];
    self.textView.text = String;
    NSParagraphStyle *style = [[NSParagraphStyle alloc]init];
    
    [self.textView insertAttributedText:attributedString settingBlock:^(NSMutableAttributedString *attributedText) {
        [attributedText addAttribute:NSParagraphStyleAttributeName value:style range:NSMakeRange(0, 20)];
    }];
    

    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: ZBSwift/KEYBOARD/KEYBOARD/ViewController.swift
//
//  ViewController.swift
//  KEYBOARD
//
//  Created by ZBin on 15/11/10.
//  Copyright © 2015年 ZBin. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITextFieldDelegate {

    
    @IBOutlet weak var mytext: UITextField!
    
    func textFieldDidBeginEditing(textField: UITextField) {
        let frame  = textField.frame
        let offset = frame.origin.y + 32 - (self.view.frame.size.height - 258)
        if offset > 0 {
            self.view.frame = CGRectMake(0.0, -offset, self.view.frame.width, self.view.frame.height)
        }
        self.view.frame = CGRectMake(0, 0, self.view.frame.width, self.view.frame.height)
    }
    
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        self.mytext.resignFirstResponder()
        return true
    }
    
    func textFieldDidEndEditing(textField: UITextField) {
        self.view.frame = CGRectMake(0, 0, self.view.frame.width, self.view.frame.height)
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        mytext.delegate = self
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

